src/inmanta/data/__init__.py

```diff
@@ -5878,27 +5958,48 @@ async def get_released_resource_details(
+            cm.date as first_generated_time,
+            rscm.model as first_model,
+            latest.model AS latest_model,
+            latest.resource_id as latest_resource_id,
+            latest.resource_type,
+            latest.agent,
+            latest.resource_id_value,
+            rps.last_deploy as latest_deploy,
+            latest.attributes,
+            {const.SQL_RESOURCE_STATUS_SELECTOR} AS status
         FROM resource first
         INNER JOIN
             /* 'latest' is the latest released version of the resource */
-            (SELECT distinct on (resource_id) resource_id, attribute_hash, model, attributes,
-                resource_type, agent, resource_id_value
-                FROM resource
-                JOIN configurationmodel cm ON resource.model = cm.version AND resource.environment = cm.environment
-                WHERE resource.environment = $1 AND resource_id = $2 AND cm.released = TRUE
-                ORDER BY resource_id, model desc
+            (SELECT distinct on (resource_id)
+                resource_id,
+                attribute_hash,
+                cm.version AS model,
+                attributes,
+                resource_type,
+                agent,
+                resource_id_value
+            FROM resource AS r
+            JOIN resource_set_configuration_model AS rscm
+                ON r.environment=rscm.environment AND r.resource_set_id=rscm.resource_set_id
+            JOIN configurationmodel AS cm
+                ON rscm.environment=cm.environment AND rscm.model=cm.version
+            WHERE r.environment=$1 AND r.resource_id=$2 AND cm.released
+            ORDER BY resource_id, model desc
             ) as latest
         /* The 'first' values correspond to the first time the attribute hash was the same as in
             the 'latest' released version */
-        ON first.resource_id = latest.resource_id AND first.attribute_hash = latest.attribute_hash
-        INNER JOIN configurationmodel cm ON first.model = cm.version AND first.environment = cm.environment
-        INNER JOIN resource_persistent_state rps on rps.resource_id = first.resource_id AND first.environment = rps.environment
-        WHERE first.environment = $1 AND first.resource_id = $2 AND cm.released = TRUE
-        ORDER BY first.resource_id, first.model asc;
+            ON first.resource_id=latest.resource_id AND first.attribute_hash=latest.attribute_hash
+        INNER JOIN resource_set_configuration_model AS rscm
+            ON first.environment=rscm.environment AND first.resource_set_id=rscm.resource_set_id
+        INNER JOIN configurationmodel AS cm
+            ON rscm.model=cm.version AND rscm.environment=cm.environment
+        INNER JOIN resource_persistent_state AS rps
+            ON rps.resource_id=first.resource_id AND first.environment=rps.environment
+        WHERE first.environment=$1 AND first.resource_id=$2 AND cm.released
+        ORDER BY first.resource_id, rscm.model asc;
```

Comment:
And this brings back my earlier concern about the index that can no longer be used for this ordering.

Reply:
We don't need to order by model right? We are doing `SELECT DISTINCT ON resource_id`

Reply:
We also order it to fetch the `latest` in this same query